### PR TITLE
[NoSpoiler] tagscript support

### PR DESCRIPTION
**Pull request by japandotorg**
test this thoroughly before merging

### CODE REVIEW

File: nospoiler/nospoiler.py [hunk 1]
```diff
@@ -323,40 +376,47 @@ async def warn(self, ctx: commands.Context) -> None:
...
+            await self.config.guild(ctx.guild).spoiler_warn_message.clear()
+            await ctx.send("Successfully resetted the warn message.")
+
+    @nospoiler.command()
+    async def delete(self, ctx: commands.Context, amount: commands.Range[int, 0, 100]):
+        """
+        Change when the warn message get's deleted.
+        """
+        await self.config.guild(ctx.guild).delete_after.set(amount)
+        await ctx.send(f"The delete after is now {amount}.")
```
Comment (ltzmax) [inline nospoiler/nospoiler.py:419]: Due to the limit from discord and avoiding rate limit, should be from  5 sec and 120 sec.

**Review by ltzmax — CHANGES_REQUESTED** [review]
Got one error in logs when having delete enabled.
```py
Oct 14 21:34:08 max python[568319]: future: <Task finished name='Task-506103' coro=<PartialMessage.delete.<locals>.delete() done, defined at /home/max/redenv/lib/python3.11/site-packages/discord/message.py:850> exception=TypeError("'<=' not supported between instances of '_ValueCtxManager' and 'int'")>
Oct 14 21:34:08 max python[568319]: Traceback (most recent call last):
Oct 14 21:34:08 max python[568319]:   File "/home/max/redenv/lib/python3.11/site-packages/discord/message.py", line 851, in delete
Oct 14 21:34:08 max python[568319]:     await asyncio.sleep(delay)
Oct 14 21:34:08 max python[568319]:   File "/usr/lib/python3.11/asyncio/tasks.py", line 634, in sleep
Oct 14 21:34:08 max python[568319]:     if delay <= 0:
Oct 14 21:34:08 max python[568319]:        ^^^^^^^^^^
Oct 14 21:34:08 max python[568319]: TypeError: '<=' not supported between instances of '_ValueCtxManager' and 'int'
```
Also have to add in info.json in requirements that tagscript is needed.

**Comment by japandotorg** [comment]
> Got one error in logs when having delete enabled.

ah forgot to await lol

**Comment by japandotorg** [comment]
should be fixed now!

File: nospoiler/nospoiler.py [hunk 2]
```diff
@@ -323,40 +378,47 @@ async def warn(self, ctx: commands.Context) -> None:
...
-
-    @nospoiler.command(aliases=["resetwarnmsg", "resetwarn"])
-    async def resetwarnmessage(self, ctx: commands.Context) -> None:
-        """Reset the spoiler warning message to default."""
-        await self.config.guild(ctx.guild).spoiler_warn_message.set(
-            "Usage of spoiler is not allowed in this server."
-        )
-        await ctx.send("Spoiler warning message has been reset.")
+    async def warnmessage(
+        self, ctx: commands.Context, *, message: TagscriptConverter
```
Comment (ltzmax) [inline nospoiler/nospoiler.py:382]: This has to be optional else it won't clear back to default.

File: nospoiler/nospoiler.py [hunk 3]
```diff
@@ -323,40 +378,47 @@ async def warn(self, ctx: commands.Context) -> None:
...
-        if spoiler_warn_message_embed:
-            await self.config.guild(ctx.guild).spoiler_warn_message_embed.set(False)
-            await ctx.send("Spoiler warning message embed is now disabled.")
         else:
-            await self.config.guild(ctx.guild).spoiler_warn_message_embed.set(True)
-            await ctx.send("Spoiler warning message embed is now enabled.")
+            await self.config.guild(ctx.guild).spoiler_warn_message.clear()
+            await ctx.send("Successfully resetted the warn message.")
+
+    @nospoiler.command()
```
Comment (ltzmax) [inline nospoiler/nospoiler.py:415]:
```suggestion
    @nospoiler.command(aliases=["timeout"])
```

**Review by ltzmax — CHANGES_REQUESTED** [review]
Mostly good, just two more things.

File: nospoiler/nospoiler.py [hunk 4]
```diff
@@ -323,40 +378,47 @@ async def warn(self, ctx: commands.Context) -> None:
...
+        **Variable:**
+        - `{server}`: [Your guild/server.](https://phen-cogs.readthedocs.io/en/latest/tags/default_variables.html#server-block)
+        - `{member}`: [Author of the message.](https://phen-cogs.readthedocs.io/en/latest/tags/default_variables.html#author-block)
+        - `{color}`: [botname]'s default color.
+
+        **Example:**
+        ```
+        {embed(description):{member(mention)} You cannot send spoilers here.}
+        {embed(color):{color}}
+        ```
```
Comment (ltzmax) [inline nospoiler/nospoiler.py:404]: I think its fair to show title too.
```suggestion
        ```
        {embed(title):No spoiler allowed.}
        {embed(description):{member(mention)} You cannot send spoilers here.}
        {embed(color):{color}}
        ```
```

**Review by ltzmax — CHANGES_REQUESTED** [review]

File: nospoiler/__init__.py [hunk 5]
```diff
@@ -1,12 +1,37 @@
...
 from redbot.core.bot import Red
+from redbot.core.errors import CogLoadError
 
 from .nospoiler import NoSpoiler
+from ._tagscript import validate_tagscriptengine
+
+VERSION_RE: Pattern[str] = re.compile(r"TagScript==(\d\.\d\.\d)")
+
+with open(Path(__file__).parent / "info.json") as f:
+    data = json.load(f)
```
Comment (ltzmax) [inline nospoiler/__init__.py:15]:
```suggestion
with open(Path(__file__).parent / "info.json") as f:
    data = orjson.loads(f.read())
```

File: nospoiler/__init__.py [hunk 6]
```diff
@@ -1,12 +1,37 @@
+import json
+import re
+from pathlib import Path
```
Comment (ltzmax) [inline nospoiler/__init__.py:3]:
```suggestion
import orjson
from pathlib import Path
```

**Review by ltzmax — CHANGES_REQUESTED** [review]
Make sure to add `orjson` into requirements too.

**Review by ltzmax — APPROVED** [review]

**Pull request merged**